lib/Backend/LowerMDShared.cpp

```diff
@@ -9151,17 +9143,30 @@ void LowererMD::GenerateFastInlineBuiltInCall(IR::Instr* instr, IR::JnHelperMeth
 
             // ROUNDSD srcCopy, srcCopy, round_mode
             IR::Opnd * roundMode;
-            if(isNotCeil)
+#ifdef ENABLE_WASM
+            if (instr->m_opcode == Js::OpCode::Trunc_A)
             {
-                roundMode = IR::IntConstOpnd::New(0x01, TyInt32, this->m_func);
+                roundMode = IR::IntConstOpnd::New(0x03, TyInt32, this->m_func);
             }
-            else if (instr->GetDst()->IsInt32() || instr->m_opcode != Js::OpCode::InlineMathFloor)
+            else if (instr->m_opcode == Js::OpCode::Nearest_A)
             {
-                roundMode = IR::IntConstOpnd::New(0x02, TyInt32, this->m_func);
+                roundMode = IR::IntConstOpnd::New(0x00, TyInt32, this->m_func);
             }
             else
+#endif //ENABLE_WASM
             {
-                roundMode = IR::IntConstOpnd::New(0x03, TyInt32, this->m_func);
+                if (isNotCeil)
```

Comment:
this is a bit confusing now. i think would be more clear what is going on if we have switch on m_opcode (and i'm pretty sure `|| instr->m_opcode != Js::OpCode::InlineMathFloor` will always be false)